fluent/syntax/ast.py

```diff
@@ -245,9 +245,9 @@ def __init__(self, id, name, **kwargs):
 
 
 class VariantExpression(Expression):
-    def __init__(self, id, key, **kwargs):
+    def __init__(self, of, key, **kwargs):
         super(VariantExpression, self).__init__(**kwargs)
-        self.id = id
+        self.of = of
```

Comment:
@Pike, @zbraniecki - do you have opinions on what we should call this? It should fit both `MessageReference` (now) as well as as `ExternalArgument` (later).

I looked up a few examples: `a.b` parses as `Attribute(value=Name(id="a"), attr: "b")` in Python and as `MemberExpression(object=Identifier(name="a"), property=Identifier(name="b"))` in JavaScript.

Reply:
no strong opinion, both Python and JS naming scheme seems better to me than `of`.

Reply:
I'd suggest `object` as it's generic enough to fit a wide range of expressions. Another idea: how about `reference`, or `ref`?

  - `-term[varname]` would parse as `VariantExpression(ref=MessageReference(id=Identifier(name="-term")), key=VariantName(name="varname"))`.

  - `$arg[varname]` would parse as `VariantExpression(ref=ExternalArgument(id=Identifier(name="-term")), key=VariantName(name="varname"))`.